app/Services/Medication/MedicationCreator.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace App\Services\Medication;
+
+use App\Models\Medication;
+use App\Repositories\Medication\MedicationRepository;
+
+
+/**
+ * Class  DoctorCreator
+ * @package App\Services\Doctor
+ */
+class MedicationCreator
+{
+     /**
+     * @var MedicationRepository
+     */
+    protected $medicationRepository;
+
+
+    /**
+     * DoctorGetter constructor.
+     * @param MedicationRepository $medicationRepository
+
+     */
+    public function __construct(MedicationRepository $medicationRepository)
+    {
+        $this->medicationRepository = $medicationRepository;
+    }
+
+    /**
+     * Store an doctor
+     * @param array $data
+     * @return \Illuminate\Database\Eloquent\Model
+     */
+    public function store(array $data)
+    {
+        $doctor =  $this->medicationRepository->store($data);
+        return $doctor->refresh();
+    }
+
+    public function bulkStore($data)
+    {
+        $medications = $data['medication'];
+
+        foreach($medications as $value){
+            $value['prescription_id'] = $data['prescription_id'];
+            $med = Medication::create($value);
+        }
```

Comment:
You can use bluk insert, use a variable or aray to store the value then after foreach loop use bulk insert, if we use this structure then in every foreach loop we communicate with the database, if there are multiple medications then this will give an error in the bulkInsert, it can be do only one communication.